crates/miden-objects/src/account/component/template/storage/mod.rs

```diff
@@ -658,6 +664,234 @@ mod tests {
+
+        let result = map_entry.try_build_storage_slots(&InitStorageData::default());
+
+        assert_matches::assert_matches!(
+            result,
+            Err(AccountComponentTemplateError::PlaceholderValueNotProvided(name))
+                if name.as_str() == "procedure_thresholds"
+        );
+
+        // try with an empty list
+
+        let init_data = InitStorageData::from_toml(
+            r#"
+            procedure_thresholds = []
+        "#,
+        )
+        .unwrap();
+
+        let result = map_entry.try_build_storage_slots(&init_data).unwrap();
+
+        assert_eq!(result.len(), 1);
+        match &result[0] {
+            StorageSlot::Map(storage_map) => assert_eq!(storage_map.num_entries(), 0),
+            _ => panic!("expected map storage slot"),
+        }
+    }
+
+    #[test]
+    fn map_placeholder_requirement_is_reported() {
+        let targets = [AccountType::RegularAccountImmutableCode].into_iter().collect();
+        let map =
+            MapRepresentation::new_template(StorageValueName::new("procedure_thresholds").unwrap())
+                .with_description("Configures procedure thresholds");
+
+        let metadata = AccountComponentMetadata::new(
+            "test".into(),
+            "desc".into(),
+            Version::new(1, 0, 0),
+            targets,
+            vec![StorageEntry::new_map(0, map)],
+        )
+        .unwrap();
+
+        let requirements = metadata.get_placeholder_requirements();
+        let requirement = requirements
+            .get(&StorageValueName::new("procedure_thresholds").unwrap())
+            .expect("map placeholder should be reported");
+
+        assert_eq!(requirement.r#type.as_str(), "map");
+        assert_eq!(requirement.description.as_deref(), Some("Configures procedure thresholds"),);
+    }
+
+    #[test]
+    fn toml_template_map_roundtrip() {
+        let toml_text = r#"
+        name = "Test Component"
+        description = "Component with templated map"
+        version = "1.0.0"
+        supported-types = ["RegularAccountImmutableCode"]
+
+        [[storage]]
+        name = "my_map"
+        description = "Some description"
+        slot = 0
+        type = "map"
+        "#;
+
+        let metadata = AccountComponentMetadata::from_toml(toml_text).unwrap();
+        assert_eq!(metadata.storage_entries().len(), 1);
+        match metadata.storage_entries().first().unwrap() {
+            StorageEntry::Map { map, .. } => match map {
+                MapRepresentation::Template { identifier } => {
+                    assert_eq!(identifier.name.as_str(), "my_map");
+                    assert_eq!(identifier.description.as_deref(), Some("Some description"));
+                },
+                MapRepresentation::Value { .. } => panic!("expected template map"),
+            },
+            _ => panic!("expected map storage entry"),
+        }
+
+        let toml_roundtrip = metadata.as_toml().unwrap();
```

Comment:
Nit: Not from this PR, but I'd consider renaming `as_toml` to `to_toml` to better follow https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv, because `as_` suggests the conversion is free, which it isn't here.

Reply:
Addressed in the last commit.